src/test/LidoJoin-integration.t.sol

```diff
@@ -0,0 +1,364 @@
+                    prevValue
+                );
+            }
+        }
+    }
+
+}
+
+// Mainnet tests against Lido rewards for Curve stETH/ETH LP
+contract LidoIntegrationTest is TestBase {
+
+    ERC20 gem;
+    ERC20 bonus;
+    StakingRewardsLike pool;
+    VatLike vat;
+    bytes32 ilk = "CURVESTETHETH-A";
+    SynthetixJoinImp join;
+    CropManagerImp manager;
+
+    Usr user1;
+    Usr user2;
+    Usr user3;
+
+    function setUp() public {
+        vat = VatLike(0x35D1b3F3D7966A1DFe207aa4514C12a259A0492B);
+        gem = ERC20(0x06325440D014e39736583c165C2963BA99fAf14E);        // Curve stETH/ETH LP
+        bonus = ERC20(0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32);      // LDO token
+        pool = StakingRewardsLike(0x99ac10631F69C753DDb595D074422a0922D9056B);
+
+        // Give this contract admin access on the vat
+        giveAuthAccess(address(vat), address(this));
+
+        CropJoin baseJoin = new CropJoin();
+        baseJoin.setImplementation(address(new SynthetixJoinImp(address(vat), ilk, address(gem), address(bonus), address(pool))));
+        join = SynthetixJoinImp(address(baseJoin));
+        join.init();
+        CropManager baseManager = new CropManager();
+        baseManager.setImplementation(address(new CropManagerImp(address(vat))));
+        manager = CropManagerImp(address(baseManager));
+        baseJoin.rely(address(manager));
+        baseJoin.deny(address(this));    // Only access should be through manager
+        vat.rely(address(baseJoin));
+        assertEq(address(join.pool()), address(pool));
+        user1 = new Usr(hevm, join, manager, gem);
+        user2 = new Usr(hevm, join, manager, gem);
+        user3 = new Usr(hevm, join, manager, gem);
+
+        assertTrue(user1.proxy() != address(0));
+        assertTrue(user2.proxy() != address(0));
+        assertTrue(user3.proxy() != address(0));
+
+        user1.giveTokens(gem, 100 ether);
+        assertEq(gem.balanceOf(address(user1)), 100 ether);
+        user2.giveTokens(gem, 100 ether);
+        user3.giveTokens(gem, 100 ether);
+    }
+
+    function test_join() public {
+        user1.join(10 ether);
+
+        assertEq(pool.balanceOf(address(join)), 10 ether);
+        assertEq(gem.balanceOf(address(join)), 0 ether);
+    }
+
+    function test_join_rewards() public {
+        user1.join(10 ether);
+
+        assertEq(pool.balanceOf(address(join)), 10 ether);
+        assertEq(gem.balanceOf(address(join)), 0 ether);
+        assertEq(bonus.balanceOf(address(join)), 0 ether);
+        assertEq(user1.bonus(), 0 ether);
+
+        // Acquire some rewards
+        hevm.warp(now + 100 days);
+        user1.reap();
+
+        assertEq(pool.balanceOf(address(join)), 10 ether);
+        assertEq(gem.balanceOf(address(join)), 0 ether);
+        assertEq(bonus.balanceOf(address(join)), 0 ether);
+        assertGt(user1.bonus(), 0 ether);
+    }
+
+    function test_join_exit() public {
```

Comment:
suggest to add also a test/check that does exit and make sure the user got bonus tokens (there is only a thet checking that the adapter got them).